askama_shared/src/path.rs

```diff
@@ -32,34 +32,23 @@ pub fn get_template_source(tpl_path: &Path) -> String {
 }
 
 pub fn find_template_from_path(path: &str, start_at: Option<&Path>) -> PathBuf {
-    let root = template_dir();
+    let dirs = template_dirs();
     if let Some(rel) = start_at {
-        let mut fs_rel_path = root.clone();
-        fs_rel_path.push(rel);
-        fs_rel_path = fs_rel_path.with_file_name(path);
+        let root = search_template_in_dirs(&dirs, rel).unwrap();
+        let fs_rel_path = root.join(rel).with_file_name(path);
```

Comment:
One question is whether the `start_at` paths here are already rooted (that is, are already absolute paths starting with the crate root) or not. Probably need to audit all uses to see whether usage is consistent or not.

Reply:
Just preserving existing behavior. If `start_at` is already rooted the relative path should not exist and it should slip through.

Reply:
Okay, but can we at least replace the `unwrap()` with an `expect()` that clearly states the problem?